tdrs-backend/tdpservice/users/permissions.py

```diff
@@ -5,9 +5,13 @@
 
 def is_own_stt(request, view):
     """Verify user belongs to requested STT."""
-    return is_in_group(request.user, "Data Prepper") and (
-        request.user.stt.id == request.data['stt']
-    )
+    specifies_stt = request.data.get('stt', False)
+    if not specifies_stt:
+        return True
+    else:
+        return is_in_group(request.user, "Data Prepper") and (
+            request.user.stt.id == request.data['stt']
+        )
```

Comment:
```suggestion
    if (requested_stt := request.data.get('stt', None)) is None:
        return True
    else:
        return is_in_group(request.user, "Data Prepper") and (
            request.user.stt.id == requested_stt
        )
```
Ultra nit, but using [assignment expressions (walrus operator)](https://www.python.org/dev/peps/pep-0572/) you can store a variable within the if conditional and re-use it later in the block to save a line of code and prevent needing to look up the value again from the request dictionary.

Reply:
Good idea